### PR TITLE
Refactor FXIOS-5626 [v111] Use URLComponents to parse mailto metadata

### PR DESCRIPTION
**Pull request by thatswinnie**
[FXIOS-5626](https://mozilla-hub.atlassian.net/browse/FXIOS-5626)
#12998

We are now using URLComponents to parse mailto links and build the link to open the correct mail app.

### CODE REVIEW

**Review by OrlaM — APPROVED** [review]

File: Client/Frontend/Browser/MailProviders.swift [hunk 1]
```diff
@@ -7,75 +7,115 @@ import Shared
...
+        if emailFormat == .userAndPassword {
+            urlComponents = addUserAndPasswordComponents(urlComponents, toQueryItem: toQueryItem)
+        }
+
+        return urlComponents
+    }
+
+    // Used to build Protonmail URL correctly (format: "protonmail://mailto:email@test.com")
+    // We use the user, password and host component parts for this URL format.
+    private func addUserAndPasswordComponents(_ components: URLComponents,
```
Comment (nbhasin2) [inline Client/Frontend/Browser/MailProviders.swift:60]: note: Is this only for proton mail then could we clean this up for proton mail

I like the comments though!! Keep them as they are helpful to understand

File: Client/Frontend/Browser/MailProviders.swift [hunk 2]
```diff
@@ -7,75 +7,115 @@ import Shared
 
 // mailto headers: subject, body, cc, bcc
 
+enum MailProviderEmailFormat {
+    case standard
+    case userAndPassword // used for Protonmail
```
Comment (nbhasin2) [inline Client/Frontend/Browser/MailProviders.swift:12]: note: lets change this to `protonMailFormat` as all this work is for proton mail

**Review by nbhasin2 — APPROVED** [review]
r+ except for some changes that we discussed

**Comment by thatswinnie** [comment]
Not running the build again for the tiny changes. Last build passed:
Screenshot 2023-01-24 at 16 26 00

**Pull request merged**